tests/functionnal/NotificationTargetTicket.php

```diff
@@ -139,5 +139,217 @@ public function testgetDataForObject() {
+         ]
+      ];
+
+      $notiftargetticket = new \NotificationTargetTicket(getItemByTypeName('Entity', '_test_root_entity', true), 'new', $ticket );
+      $ret = $notiftargetticket->getDataForObject($ticket, $basic_options);
+
+      //get all task / solution / followup (because is tech)
+      $expected = [
+         [
+            "##timelineitems.type##"        => "TicketTask",
+            "##timelineitems.typename##"    => "Ticket tasks",
+            "##timelineitems.date##"        => $task_tech->fields['date'],
+            "##timelineitems.description##" => $task_tech->fields['content'],
+            "##timelineitems.position##"    => "right",
+            "##timelineitems.author##"      => "tech",
+         ],[
+            "##timelineitems.type##"        => "TicketTask",
+            "##timelineitems.typename##"    => "Ticket tasks",
+            "##timelineitems.date##"        => $task_private->fields['date'],
+            "##timelineitems.description##" => $task_private->fields['content'],
+            "##timelineitems.position##"    => "right",
+            "##timelineitems.author##"      => "tech",
+         ],[
+            "##timelineitems.type##"        => "ITILSolution",
+            "##timelineitems.typename##"    => "Solutions",
+            "##timelineitems.date##"        => $solution->fields['date_creation'],
+            "##timelineitems.description##" => $solution->fields['content'],
+            "##timelineitems.position##"    => "right",
+            "##timelineitems.author##"      => "", //empty
+         ],[
+            "##timelineitems.type##" => "ITILFollowup",
+            "##timelineitems.typename##"=> "Followups",
+            "##timelineitems.date##"        => $fup_private_tech->fields['date'],
+            "##timelineitems.description##" => $fup_private_tech->fields['content'],
+            "##timelineitems.position##"=> "right",
+            "##timelineitems.author##"=> "tech",
+         ],[
+            "##timelineitems.type##"        => "ITILFollowup",
+            "##timelineitems.typename##"    => "Followups",
+            "##timelineitems.date##"        => $fup_post_only->fields['date'],
+            "##timelineitems.description##" => $fup_post_only->fields['content'],
+            "##timelineitems.position##"    => "left",
+            "##timelineitems.author##"      => "post-only",
+         ],[
+            "##timelineitems.type##" => "ITILFollowup",
+            "##timelineitems.typename##" => "Followups",
+            "##timelineitems.date##"        => $fup_tech->fields['date'],
+            "##timelineitems.description##" => $fup_tech->fields['content'],
+            "##timelineitems.position##" => "right",
+            "##timelineitems.author##" => "tech",
+         ]
+      ];
+
+      $this->array($ret['timelineitems'])->isIdenticalTo($expected);
+
+      $auth = new \Auth();
+      $this->boolean((boolean)$auth->login('post-only', 'postonly', true))->isTrue();
```

Comment:
Why not using `$this->login()`?

Reply:
bad copy/paste without understanding
solved